src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -1138,4 +1138,422 @@ public static int MoveNansToFront<TKey, TValue>(Span<TKey> keys, Span<TValue> va
             return left;
         }
     }
+
+    #region ArraySortHelper for TComparer
+
+    #region ArraySortHelper for single arrays
+
+    internal sealed partial class ArraySortHelper<T>
```

Comment:
Can this be `ArraySortHelperForTComparer<T, TComparer> where TComparer : IComparer<T>` so that methods are non-generic?

Reply:
One generic method calling another generic method has extra overhead in shared generic code compared to one method calling another method on the same generic type.

Yes, a single generic class instantiations consumes more than a single generic method. We have one generic class instantiation vs. a bunch of generic method instantiations here. My guess is that the memory consumption of two variants is close.